app/controllers/organisation_relationships_controller.rb

```diff
@@ -31,18 +33,59 @@ def managing_agents
     end
   end
 
+  def add_housing_provider
+    @organisations = Organisation.where.not(id: @organisation.id).pluck(:id, :name)
+    respond_to do |format|
+      format.html do
+        render "organisation_relationships/add_housing_provider", layout: "application"
+      end
+    end
+  end
+
+  def create_housing_provider
+    child_organisation_id = @organisation.id
+    parent_organisation_id = related_organisation_id
+    relationship_type = OrganisationRelationship::OWNING
+    if related_organisation_id.empty?
+      @organisation.errors.add :related_organisation_id, "You must choose a housing provider"
+      @organisations = Organisation.where.not(id: child_organisation_id).pluck(:id, :name)
+      render 'organisation_relationships/add_housing_provider'
+      return
+    elsif OrganisationRelationship.exists?(child_organisation_id:, parent_organisation_id:, relationship_type:)
+      @organisation.errors.add :related_organisation_id, "You have already added this housing provider"
+      @organisations = Organisation.where.not(id: child_organisation_id).pluck(:id, :name)
+      render 'organisation_relationships/add_housing_provider'
+      return
+    end
+    create(child_organisation_id:, parent_organisation_id:, relationship_type:)
+    redirect_to housing_providers_organisation_path(related_organisation_id:)
+  end
+
+  def create_managing_agent
+    create(child_organisation_id: related_organisation_id, parent_organisation_id: @organisation.id, relationship_type: 1)
```

Comment:
let's use the constants rather than the numbers

`OrganisationRelationship::OWNING`